src/common-events/utils/headless-cms/service.tsx

```diff
@@ -94,24 +96,26 @@ export const getAllPages = async (): Promise<PageInfo[]> => {
         locale: node.language.code.toLowerCase(),
         slug: node.slug,
       });
-      node.translations?.forEach((translation) => {
-        if (
-          translation?.uri &&
-          translation.slug &&
-          translation.language?.code
-        ) {
-          const {
-            uri,
-            slug,
-            language: { code },
-          } = translation;
-          pageInfos.push({
-            uri,
-            slug,
-            locale: code.toLocaleLowerCase(),
-          });
+      node.translations?.forEach(
+        (translation: PageType['translation'][number]) => {
```

Comment:
PageType['translation'][number] is actually a PageType, but since all the node fields are interpreted as any type fields, I think I can pass this this time 😛 . But, the next time when I have a problem with this, I call you to fix it 😎 Using the PageType seems to give a better hint for IDE about the content of the node.

Reply:
![image](https://user-images.githubusercontent.com/389204/189633380-ad41bcc8-1544-47d3-8ad5-fc3d9bc83184.png)

strange thing